python/pylibcudf/pylibcudf/tests/test_partitioning.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+import pyarrow as pa
+import pylibcudf as plc
+import pytest
+
+
+@pytest.fixture
+def pa_table():
+    return pa.table({"a": [1, 2, 3], "b": [1, 2, 5], "c": [1, 2, 10]})
+
+
+def test_partition(pa_table):
+    plc_result, result_offsets = plc.partitioning.partition(
+        plc.interop.from_arrow(pa_table),
```

Comment:
We generally prefer putting both the arrow object and the corresponding cudf object into a combined fixture and making it have module scope. That minimizes the number of distinct GPU allocations required to run the test suite. [Here's an example](https://github.com/rapidsai/cudf/blob/branch-24.10/python/pylibcudf/pylibcudf/tests/test_reshape.py#L9). [Some older tests](https://github.com/rapidsai/cudf/blob/branch-24.10/python/pylibcudf/pylibcudf/tests/test_binaryops.py) will set those up as two separate fixtures, with the [pylibcudf fixture](https://github.com/rapidsai/cudf/blob/branch-24.10/python/pylibcudf/pylibcudf/tests/test_binaryops.py#L71) depending on the [pyarrow fixture](https://github.com/rapidsai/cudf/blob/branch-24.10/python/pylibcudf/pylibcudf/tests/test_binaryops.py#L64). That is also OK, but overall the combined fixture is a bit cleaner.